scipy/optimize/_differentialevolution.py

```diff
@@ -116,12 +116,32 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
         Specify `seed` for repeatable minimizations.
     disp : bool, optional
         Prints the evaluated `func` at every iteration.
-    callback : callable, `callback(xk, convergence=val)`, optional
-        A function to follow the progress of the minimization. ``xk`` is
-        the best solution found so far. ``val`` represents the fractional
-        value of the population convergence.  When ``val`` is greater than one
-        the function halts. If callback returns `True`, then the minimization
-        is halted (any polishing is still carried out).
+    callback : callable, optional
+        A callable called after each iteration. Has the signature:
+
+            ``callback(intermediate_result: OptimizeResult)``
+
+        where ``intermediate_result`` is a keyword parameter containing an
+        `OptimizeResult` with attributes ``x`` and ``fun``, the best solution
+        found so far and the objective function. Note that the name
+        of the parameter must be ``intermediate_result`` for the callback
+        to be passed an `OptimizeResult`.
+
+        The callback also supports a signature like:
+
+            ``callback(x, convergence: float =val)``
+
+        ``val`` represents the fractional value of the population convergence.
+        When ``val`` is greater than one the function halts.
+
+        Introspection is used to determine which of the signatures is invoked.
+
+        The minimization will halt if the callback raises ``StopIteration``
+        or returns ``True`` (any polishing is still carried out).
```

Comment:
```suggestion
        Global minimization will halt if the callback raises ``StopIteration``
        or returns ``True``. Local minimization (polishing) is still carried out.
```

Reply:
I'm not a fan of the Global and Local words here, because it may lead people to believe that local minimisation is carried out multiple times, a.la. shgo/basinhopping.

Reply:
Ok. My concern here is that it is ambiguous whether "minimization" refers to DE or also includes a polishing step. There was conflicting information on this point elsewhere. No need to accept the suggestion as written, but please find a way to make this point more explicit.